File: 09.Greed贪心/leetcode_3702_longestSubsequence.cpp
#include <iostream>
#include <vector>
using namespace std;

class Solution {
public:
    int longestSubsequence(vector<int>& nums) {
        int tot = 0;
        bool allZero = true;
        for (int i : nums){
            tot ^= i;
            allZero &= (i==0);
        }
        int n = nums.size();
        if (tot == 0){
            return allZero ? 0 : n-1;
        }
        return n;
    }
};

int main(){
    Solution s;
    vector<int> nums = {1,2,3};
    cout << s.longestSubsequence(nums) << endl;
}

File: 09.Greed贪心/leetcode_3702_longestSubsequence.py
from typing import List

class Solution:
    def longestSubsequence(self, nums: List[int]) -> int:
        n = len(nums)
        tot = 0
        allZero = True
        for i in nums:
            tot ^= i
            allZero &= (i == 0)

        if tot == 0:
            if allZero:
                return 0
            return n-1
        return n

if __name__ == "__main__":
    s = Solution()
    nums = [1,2,3]
    print(s.longestSubsequence(nums))
